src/game/AI/ScriptDevAI/scripts/outland/hellfire_peninsula.cpp

```diff
@@ -48,69 +48,43 @@ EndContentData */
 
 enum
 {
-    SAY_FREE                        = -1000139,
-
-    FACTION_HOSTILE                 = 16,
     FACTION_FRIENDLY                = 35,
-
-    SPELL_ENVELOPING_WINDS          = 15535,
-    SPELL_SHOCK                     = 12553,
+    AERANAS_RELAY_SCRIPT            = 17086,
 };
 
-struct npc_aeranasAI : public ScriptedAI
+enum AeranasActions
 {
-    npc_aeranasAI(Creature* pCreature) : ScriptedAI(pCreature) { Reset(); }
-
-    uint32 m_uiEnvelopingWindsTimer;
-    uint32 m_uiShockTimer;
+    AERANAS_PHASE_1,
+    AERANAS_ACTION_MAX,
+};
 
-    void Reset() override
+struct npc_aeranasAI : public CombatAI
+{
+    npc_aeranasAI(Creature* creature) : CombatAI(creature, AERANAS_ACTION_MAX)
     {
-        m_uiEnvelopingWindsTimer = 9000;
-        m_uiShockTimer           = 5000;
-
-        m_creature->RemoveFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_QUESTGIVER);
+        AddTimerlessCombatAction(AERANAS_PHASE_1, true);
+        Reset(); 
     }
 
-    void UpdateAI(const uint32 uiDiff) override
+    void ExecuteAction(uint32 action) override
     {
-
-        if (!m_creature->SelectHostileTarget() || !m_creature->GetVictim())
-            return;
-
-        if (m_creature->GetHealthPercent() < 30.0f)
-        {
-            m_creature->SetFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_QUESTGIVER);
-            m_creature->RemoveAllAuras();
-            m_creature->CombatStop(true);
-            DoScriptText(SAY_FREE, m_creature);
-            return;
-        }
-
-        if (m_uiShockTimer < uiDiff)
-        {
-            DoCastSpellIfCan(m_creature->GetVictim(), SPELL_SHOCK);
-            m_uiShockTimer = 10000;
-        }
-        else
-            m_uiShockTimer -= uiDiff;
-
-        if (m_uiEnvelopingWindsTimer < uiDiff)
+        switch (action)
         {
-            DoCastSpellIfCan(m_creature->GetVictim(), SPELL_ENVELOPING_WINDS);
-            m_uiEnvelopingWindsTimer = 25000;
+            case AERANAS_PHASE_1:
+            {
+                if (m_creature->GetHealthPercent() < 30.0f) 
+                {
+                    m_creature->GetMap()->ScriptsStart(SCRIPT_TYPE_RELAY, AERANAS_RELAY_SCRIPT, m_creature, m_creature);
```

Comment:
Missing a DisableCombatAction(action); here, I know you are doing combat stop, buuut just in case something would put it in combat immediately again

Reply:
Done